shimmingtoolbox/dicom_to_nifti.py

```diff
@@ -0,0 +1,128 @@
+import os
+import numpy as np
+from .read_nii import read_nii
+
+
+def dicom_to_nifti(unsortedDicomDir, niftiPath):
+
+    #os.mkdir(niftiPath)
+    print(unsortedDicomDir)
+    print(niftiPath)
+
+    # Check for which "which" to use depending on OS
+    if os.name == 'nt':  # If OS is Windows
+        which = 'where'
+        copy = 'copy'
+    else:
+        which = 'which'
+        copy = 'cp -r'
+
+    # Make sur dcm2niix is installed
+    if os.system(which + ' dcm2niix') != 0:
+        print('Error: dcm2niix is not installed.')
+
+    # Make sure dcm2bids is installed
+    if os.system(which + ' dcm2bids') != 0:
+        raise Exception('Error: dcm2bids is not installed.')
```

Comment:
Exceptions are wuite powerfull, but you should look in the Buil-in libraries to raise a more specific exception type (specially since you are using os calls) like OSError (FileExistsError, IsADirectoryError, etc) for folder checks and include exceptions. You can find all built-in excpetions here: [https://docs.python.org/3/library/exceptions.html#os-exceptions](url)

Reply:
I wouldn't bother raising an exception here at all, to be honest. Instead, use [`subprocess.run`](
https://docs.python.org/3/library/subprocess.html#subprocess.run); it will take care of raising an appropriate exception.

In general python code is exception-heavy. My philosophy is to think as little about exceptions as possible in python and not try to second guess deeper code. It will usually give you a useful message when it really can't continue.

```
subprocess.run(["dcm2bids", "-i", ...], check=True)  # check means "raise an exception if dcm2bids fails for any reason, including for it not being installed"
```

Reply:
i think we could also use dcm2bids as a python package directly